### DISCUSSION TITLE
Strongswan kernel HA patch

### DISCUSSION THREAD

**Discussion by muvarov** [strongswan/strongswan#2416] [General]
Hello, does anybody know the story about HA kernel patch? Why this patch is maintaining separately? Did somebody try to upstream it? Were there issues or discussions about it?

Thanks,
Maxim.

**Comment by muvarov** [comment]
I found git commit which removes it:
commit 9db5d918e2c07fa09fab18bc7addf3408da0c76f
Author: Florian Westphal <fw@strlen.de>
Date:   Thu Jan 5 20:22:02 2023 +0100

    netfilter: ip_tables: remove clusterip target
    
    Marked as 'to be removed soon' since kernel 4.1 (2015).
    Functionality was superseded by the 'cluster' match, added in kernel
    2.6.30 (2009).
    
    clusterip_tg_check still has races that can give
    
     proc_dir_entry 'ipt_CLUSTERIP/10.1.1.2' already registered
    
    followed by a WARN splat.
    
    Remove it instead of trying to fix this up again.
    clusterip uapi header is left as-is for now.
    
    Signed-off-by: Florian Westphal <fw@strlen.de>

And patchset: https://lore.kernel.org/netdev/20230118123208.17167-1-fw@strlen.de/T/

Which says "Remove the CLUSTERIP target.  Marked as obsolete for years,
   and we still have WARN splats wrt. races of the out-of-band
   /proc interface installed by this target."

That is not really clear for me, instead of fixing it upsteam this patch now maintained inside strongswan? 

Upper commit is about net/ipv4/netfilter/ipt_CLUSTERIP.c module. What about others xfrm changes?

**Comment by tobiasbrunner** [comment]
The [HA patch](https://docs.strongswan.org/docs/5.9/features/highAvailability.html) relies on the deprecated and eventually removed CLUSTERIP module (which is also why it only support IPv4). Since its removal from the kernel sources, the latter is now part of the patch as it won't work without it. No part of the HA patches were ever upstreamed (don't think it was ever tried either).
